File: backend/routers/gmail.py
from fastapi import APIRouter, Depends, Request
from supabase import Client
from services.gmail_service import test_gmail_connection
from core.database import get_supabase
from core.logger import get_logger

router = APIRouter(prefix="/gmail", tags=["gmail"])
logger = get_logger(__name__)

@router.get("/test/{user_id}")
def test_connection(user_id: str, supabase: Client = Depends(get_supabase)):
    return test_gmail_connection(supabase, user_id)

@router.post("/save-refresh-token")
async def gmail_token(req: Request):
    data = await req.json()
    
    return {"message": data}

File: app/actions/gmail.ts
import { api } from "@/lib/api";

export async function testGmailConnection(userId: string) {
  return fetch(api(`/gmail/test/${userId}`)).then((r) => r.json());
}

export async function saveRefreshToken(user_id: string, refresh_token: string) {
  return fetch(api("/gmail/save-refresh-token"), {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ user_id, refresh_token }),
  }).then((r) => r.json());
}
